### PR TITLE
[MER-2731] Remove max limits on Datashop download and add caching

### PR DESCRIPTION
**Pull request by gastonabella**
[MER-2731](https://eliterate.atlassian.net/browse/MER-2731)

Second PR for improving the Datashop export feature (first PR https://github.com/Simon-Initiative/oli-torus/pull/4459)

It introduces the following changes:
- Process the Datashop export process in batches, showing the number of batches processed in real time.
- Use a Cache for storing intermediate data.
- Some UI/CSS improvements.

https://github.com/Simon-Initiative/oli-torus/assets/26532202/46d4d78c-4699-4bec-8a49-59c0d499625e

[MER-2731]: https://eliterate.atlassian.net/browse/MER-2731?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -335,4 +335,6 @@ if config_env() == :prod do
             ]
         end
     ]
+
+  config :oli, :datashop, cache_limit: System.get_env("DATASHOP_CACHE_LIMIT", "200")
```
Comment (darrensiegel) [inline config/runtime.exs:339]: There is a missing call to `String.to_integer` here

File: lib/oli/analytics/datashop_export_worker.ex [hunk 2]
```diff
@@ -121,4 +89,45 @@ defmodule Oli.Analytics.DatashopExportWorker do
...
+    if batch_count != 0 do
+      Enum.reduce(1..batch_count, 0, fn batch_index, offset ->
+        # notify subscribers that a new batch has started
+        Broadcaster.broadcast_datashop_export_batch_started(
+          project.slug,
+          batch_index,
+          batch_count
+        )
+
+        Datashop.build_context(project.id, section_ids)
```
Comment (darrensiegel) [inline lib/oli/analytics/datashop_export_worker.ex:110]: The call to `Datashop.build_context` can be done one time, outside of this `Enum.reduce`.  That will limit the number of queries that the Datashop export makes

**Review by darrensiegel — CHANGES_REQUESTED** [review]
Really nice work, the caching is going to have a big impact on performance here.  Just a couple of things that need to be fixed

File: lib/oli/analytics/datashop_export_worker.ex (hunk 2 above)
Comment (gastonabella) [reply lib/oli/analytics/datashop_export_worker.ex]: Makes sense! The last commit includes these changes :)

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**